lms/lmsdb/models.py

```diff
@@ -209,6 +220,25 @@ def on_save_handler(model_class, instance, created):
         instance.api_key = generate_password_hash(instance.api_key)
 
 
+class UserCourse(BaseModel):
+    user = ForeignKeyField(User, backref='usercourses')
+    course = ForeignKeyField(Course, backref='usercourses')
+    date = DateTimeField(default=datetime.now)
+
+    @classmethod
+    def is_user_registered(cls, user_id: int, course_id: int) -> bool:
+        return (
+            cls.
+            select()
+            .join(User)
+            .where(User.id == user_id)
+            .switch()
+            .join(Course)
+            .where(Course.id == course_id)
+            .exists()
+        )
```

Comment:
Why not just taking it from this table? ^^" It might be much cheaper (`SELECT EXISTS(*) FROM UsersCourses WHERE course == course_id && user == user_id` won't work?)